---
title: "integr_scDART"
output: html_document
---

In this document we do the integration of the YTH and YTH-mut scDART data using the Seurat package. 
We also determine the cell clusters and cell types present in the data and the markers for each cell type, aided by the tools scibet and scMCA.
Finally we plot some RNA density figures shown in the article.

```{r, echo = FALSE, warning=FALSE, message = FALSE}
library(Seurat)
library(tidyverse)

# Filenames
yth_filename <- "/GPFS/Magda_lab_temp/maitenat/scs/6J-3M-2ws-V2-3-YTH_RSEC_MolsPerCell.csv"
ythmut_filename <- "/GPFS/Magda_lab_temp/maitenat/scs/6J-3M-V2-3-mut_RSEC_MolsPerCell.csv"

# Initialize the Seurat object with the raw (non-normalized data)
yth_mat <- read.csv(yth_filename, skip = 6, row.names = 1)
yth_count_mat <- t(yth_mat)
rownames(yth_count_mat) <- str_replace(rownames(yth_count_mat),'[.]','-')
scRNA_yth <- CreateSeuratObject(counts = yth_count_mat, assay = "RNA", project = '6J_yth')
#
ythmut_mat <- read.csv(ythmut_filename, skip = 6, row.names = 1)
ythmut_count_mat <- t(ythmut_mat)
rownames(ythmut_count_mat) <- str_replace(rownames(ythmut_count_mat),'[.]','-')
scRNA_ythmut <- CreateSeuratObject(counts = ythmut_count_mat, assay = "RNA", project = '6J_ythmut')

# Mitochondrial & ribosomal
scRNA_yth[["percent.mt"]] <- PercentageFeatureSet(scRNA_yth, pattern = "^mt-")
scRNA_yth[["percent.rb"]] <- PercentageFeatureSet(scRNA_yth, pattern = "^n-R5")

# Mitochondrial & ribosomal
scRNA_ythmut[["percent.mt"]] <- PercentageFeatureSet(scRNA_ythmut, pattern = "^mt-")
scRNA_ythmut[["percent.rb"]] <- PercentageFeatureSet(scRNA_ythmut, pattern = "^n-R5")
```

## Standard pre-processing workflow

The steps below encompass the standard pre-processing workflow for scRNA-seq data in Seurat. These represent the selection and filtration of cells based on QC metrics, data normalization and scaling, and the detection of highly variable features.

A few QC metrics commonly used include: 

1. The number of unique genes detected in each cell.

- Low-quality cells or empty droplets will often have very few genes

- Cell doublets or multiplets may exhibit an aberrantly high gene count

2. Similarly, the total number of molecules detected within a cell (correlates strongly with unique genes)

3. The percentage of reads that map to the mitochondrial genome

- Low-quality / dying cells often exhibit extensive mitochondrial contamination

Let's see these metrics. The left plot corresponds to the number of unique genes; middle-left to the number of molecules per cell, the middle-right percentage of reads that map to the mitochondrial genome and the rightest percentage of ribosomal proteins.


```{r, echo = FALSE, warning=FALSE, message = FALSE}
# Visualize QC metrics as a violin plot
# Left is number of unique genes per cell, middle-left number of molecules per cell, middle-right percentage of reads that map to the mitochondrial genome and rightest percentage of ribosomal proteins
print(cat("YTH sample"))
VlnPlot(scRNA_yth, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.rb"), ncol = 4, pt.size = 0)
print(cat("YTH-mut sample"))
VlnPlot(scRNA_ythmut, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.rb"), ncol = 4, pt.size = 0)
```

We will now filter cells that have unique feature counts over 6000 or less than 500 and cells that have >10% mitochondrial counts, and visualize the same plots with these data:

```{r, echo = FALSE, warning=FALSE, message = FALSE}
print(cat("YTH sample"))
scRNA_yth_filt <- subset(scRNA_yth, subset = nFeature_RNA > 500 & nFeature_RNA < 6000 & percent.mt <10)
VlnPlot(scRNA_yth_filt, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.rb"), ncol = 4, pt.size = 0)
print(cat("YTH-mut sample"))
scRNA_ythmut_filt <- subset(scRNA_ythmut, subset = nFeature_RNA > 500 & nFeature_RNA < 6000 & percent.mt <10)
VlnPlot(scRNA_ythmut_filt, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.rb"), ncol = 4, pt.size = 0)
```

## Before integration

```{r, echo = FALSE, warning=FALSE, message = FALSE}
# Merge into one seurat object
merged_seurat <- merge(scRNA_yth_filt, y = scRNA_ythmut_filt, add.cell.ids = c("yth", "yth-mut"), project = "dartseq")
# Normalize
merged_seurat <- NormalizeData(merged_seurat)
merged_seurat <- FindVariableFeatures(merged_seurat, selection.method = "vst", nfeatures = 2000)
all.genes <- rownames(merged_seurat)
merged_seurat <- ScaleData(merged_seurat, features = all.genes)
merged_seurat <- RunPCA(merged_seurat, features = VariableFeatures(object = merged_seurat)) # the features = VariableFeatures makes the PCA to use only the 2000 most variable features we identified in the previous steps
ElbowPlot(merged_seurat, ndims = 40)
```

We will choose 30 principal components.

```{r, echo = FALSE, warning=FALSE, message = FALSE}
merged_seurat <- FindNeighbors(merged_seurat, dims = 1:30)
merged_seurat <- FindClusters(merged_seurat, resolution = 0.5) 
table(merged_seurat@meta.data$seurat_clusters)
merged_seurat <- RunUMAP(merged_seurat, dims = 1:30)
DimPlot(merged_seurat, reduction = "umap", group.by = "orig.ident")
```

The clusters of yth and yth-mut are very similar, and it does not seem to be any clustering due to this condition - this suggests that the variability we see is biological and it is not due to technical issues, i.e., batch effect (if the latter was the case, we would be seing different clusters for each condition).

Even though there is no apparent batch effect, we'll perform the steps that are usually carried out for integration and the removal of batch effects.

## After integration

```{r, echo = FALSE, warning=FALSE, message = FALSE}
dart_list <- list()
dart_list[["yth"]] <- scRNA_yth_filt
dart_list[["yth_mut"]] <- scRNA_ythmut_filt

for (i in 1:length(dart_list)) {
  dart_list[[i]] <- NormalizeData(dart_list[[i]], verbose = F)
  dart_list[[i]] <- FindVariableFeatures(dart_list[[i]], selection.method = "vst", nfeatures = 2000, verbose = F)
}

# select integration anchors
anchors <- FindIntegrationAnchors(object.list = dart_list, dims = 1:30)
integrated_seurat <- IntegrateData(anchorset = anchors, dims = 1:30)
DefaultAssay(integrated_seurat) <- "integrated" # just in case. if this is set to RNA, we work with un-integrated data
integrated_seurat <- ScaleData(integrated_seurat, verbose = F)
integrated_seurat <- RunPCA(integrated_seurat, npcs = 30, verbose = F)
integrated_seurat <- RunUMAP(integrated_seurat, reduction = "pca", dims = 1:30, verbose = F)
DimPlot(integrated_seurat, reduction = "umap")
DimPlot(integrated_seurat, reduction = "umap", split.by = "orig.ident") + NoLegend()
```

Data looks completely integrated, so we can relate cell clusters in each condition! Let's now look at the clusters in the integrated data.

```{r, echo = FALSE, warning=FALSE, message = FALSE}
integrated_seurat <- FindNeighbors(integrated_seurat, dims = 1:30, verbose = F)
integrated_seurat <- FindClusters(integrated_seurat, verbose = F)
DimPlot(integrated_seurat, label = T) + NoLegend()
```


### Cell type markers based on differential expression

Note that the differential expression analysis here looks for differentialy expressed genes that are conserved across the yth and yth-mut samples.

```{r, echo = FALSE, warning=FALSE, message = FALSE}
get_conserved <- function(cluster){
        FindConservedMarkers(integrated_seurat,
                             ident.1 = cluster,
                             grouping.var = "orig.ident",
                             only.pos = TRUE,
                             min.pct = 0.1, 
                             logfc.threshold = 0.3) %>%
          #filter(`6J_yth_p_val_adj` <= 0.05) %>%
          rownames_to_column(var = "gene") %>%
          cbind(cluster_id = cluster, .)
}

n_clusters <- length(unique(integrated_seurat@meta.data$seurat_clusters))
conserved_markers <- map_dfr(setdiff(0:(n_clusters-1), 13), get_conserved)

conserved_13 <- FindConservedMarkers(integrated_seurat,
                             ident.1 = 13,
                             grouping.var = "orig.ident",
                             only.pos = TRUE,
                             min.pct = 0.1, 
                             logfc.threshold = 0.1) %>%
          rownames_to_column(var = "gene") %>%
          cbind(cluster_id = 13, .)

all_df <- bind_rows(conserved_markers, conserved_13)
write.table(all_df, file = "scDART_DEgenes.csv", col.names = TRUE, quote = FALSE, row.names = FALSE, sep = "\t")

```


### Confounder plots per cluster

In this section I plot confounder values per cluster.

```{r, echo = FALSE, warning=FALSE, message = FALSE}
integrated_seurat$orig.ident <- factor(x = integrated_seurat$orig.ident, levels = unique(integrated_seurat$orig.ident))
VlnPlot(integrated_seurat, features = "percent.mt", pt.size = 0, split.by = "orig.ident", split.plot = TRUE, col = c("#F8766D", "#00BFC4")) 
```

```{r, echo = FALSE, warning=FALSE, message = FALSE}
VlnPlot(integrated_seurat, features = "percent.rb", pt.size = 0, split.by = "orig.ident", split.plot = TRUE, col = c("#F8766D", "#00BFC4"))  
```

```{r, echo = FALSE, warning=FALSE, message = FALSE}
VlnPlot(integrated_seurat, features = "nCount_RNA", pt.size = 0, split.by = "orig.ident", split.plot = TRUE, col = c("#F8766D", "#00BFC4"))  
VlnPlot(integrated_seurat, features = "nFeature_RNA", pt.size = 0, split.by = "orig.ident", split.plot = TRUE, col = c("#F8766D", "#00BFC4")) 
```


## Cluster final plot

Here I plot the cluster plot with the names of the cell types that identify each cluster.
This has been aided by the tools scibet and scMCA. The scripts for running these in this same folder, namely scibet_tm.R and scMCA.R, respectively.


### 5 groups:

```{r, echo = FALSE, warning=FALSE, message = FALSE}
oligodendrocytes <- c(2, 3, 5, 6, 7, 12, 13, 16)
neurons <- c(11, 23)
endothelial_cells <- c(26)
immune_cells <- c(0, 1, 4, 8, 9, 10, 14, 15, 17, 18, 19, 20, 21, 24, 25, 27)
astrocytes <- c(22)
label_mat <- cbind(c(rep("Oligodendrocyte lineage", length(oligodendrocytes)), 
                     rep("Neuronal lineage", length(neurons)),
                     rep("Endothelial cells", length(endothelial_cells)),
                     rep("Immune cells", length(immune_cells)),
                     rep("Astrocytes", length(astrocytes))),
                   c(oligodendrocytes, neurons, endothelial_cells, immune_cells, astrocytes))

int_vec <- integrated_seurat$integrated_snn_res.0.8
int_vec2 <- rep(NA, length(int_vec))

for (x in unique(label_mat[,1])){
  cl.x <- label_mat[label_mat[,1] == x,2]
  int_vec2[which(int_vec %in% cl.x)] <- x
}

integrated_seurat$cluster <- int_vec2

integrated_seurat$cluster <- factor(integrated_seurat$cluster, levels = c("Oligodendrocyte lineage", "Neuronal lineage", "Endothelial cells", "Immune cells", "Astrocytes"))

DimPlot(integrated_seurat, group.by = "cluster", label = T) + NoLegend()
DimPlot(integrated_seurat, group.by = "cluster", label = FALSE)
DimPlot(integrated_seurat, group.by = "cluster", label = T , repel = T, label.size = 3) + NoLegend()
```

Number of cells in each cluster:

```{r, echo = FALSE, warning=FALSE, message = FALSE}
cell_table <- table(integrated_seurat@meta.data$seurat_clusters, integrated_seurat@meta.data$orig.ident)
cell_table

cell_table_melt <- reshape2::melt(cell_table)
ggplot(cell_table_melt, aes(x = Var1, y = value, fill = Var2)) + 
geom_bar(stat = "identity", position = "dodge") + 
theme_bw() +
labs(x = "Cluster", y = "Number of cells per cluster")
```

Number of cells in each cell type:

```{r, echo = FALSE, warning=FALSE, message = FALSE}
cell_table <- table(integrated_seurat@meta.data$cluster, integrated_seurat@meta.data$orig.ident)
cell_table

cell_table_melt <- reshape2::melt(cell_table)
ggplot(cell_table_melt, aes(x = Var1, y = value, fill = Var2)) + 
geom_bar(stat = "identity", position = "dodge") + 
theme_bw() +
labs(x = "Cluster", y = "Number of cells per cell type")
```


Detected genes per cluster and cell type:

```{r, echo = FALSE, warning=FALSE, message = FALSE, dev=c('png', 'pdf')}
nFeat_df <- data.frame(cluster = integrated_seurat@meta.data$seurat_clusters, 
                       cell_type = integrated_seurat@meta.data$cluster,
                       nFeatures = integrated_seurat@meta.data$nFeature_RNA)

nFeat_max_df <- nFeat_df %>%
  group_by(cell_type) %>%
  summarize(max_nFeatures = max(nFeatures))

nFeat_plot <- ggplot(nFeat_max_df, aes(x = cell_type, y = max_nFeatures)) +
  geom_bar(stat = "identity", aes(fill = cell_type)) +
  coord_flip() +
  theme_classic() +
  xlab("") +
  ylab("") +
  theme(legend.position="none")

nFeat_plot

cat("Number of (max) detected genes per cluster:")
print(nFeat_max_df)

nFeat_cluster_max_df <- nFeat_df %>%
  group_by(cluster) %>%
  summarize(max_nFeatures = max(nFeatures))

```


```{r, echo = FALSE, warning=FALSE, message = FALSE}
VlnPlot(integrated_seurat,
        ncol = 1, 
        features = c("Sox10", "Cldn11", "Mbp", "Lcp1", "Arpc2", "Arf6", "Snap25", "Atp1a3", "Clstn2", "Gja1", "Gfap", "Aldh1a1", "Esam", "Ly6c1", "Itm2a"), 
        pt.size = 0) & theme(axis.title.x = element_blank(),
                             axis.title.y = element_blank(),
                             title = element_text(size = 6))

Nebulosa::plot_density(integrated_seurat, features = c("Sox10", "Snap25", "Esam", "Lcp1", "Gfap"), joint = FALSE, combine = FALSE)

Nebulosa::plot_density(integrated_seurat, "Camk2a", joint = FALSE, combine = FALSE)
```

Conserved m6a RNA density plots:
```{r, echo = FALSE, warning=FALSE, message = FALSE}
Nebulosa::plot_density(integrated_seurat, "Smarcc2", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Apoe", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Aplp1", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Ctsb", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Itm2b", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Mecp2", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Syt11", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Lamp1", joint = FALSE, combine = FALSE)
Nebulosa::plot_density(integrated_seurat, "Brd2", joint = FALSE, combine = FALSE)
```

```{r, echo = FALSE, warning=FALSE, message = FALSE}
Nebulosa::plot_density(integrated_seurat, "App", joint = FALSE, combine = FALSE)
```

m6a regulatory enzymes:

```{r, echo = FALSE, warning=FALSE, message = FALSE}
Nebulosa::plot_density(integrated_seurat, features = c("Ythdf2", "Mettl3", "Mettl14", "Wtap", "Fto", "Alkbh5"), joint = FALSE, combine = FALSE)
```


Save seurat object:

```{r, echo = FALSE, warning=FALSE, message = FALSE}
saveRDS(integrated_seurat, file = "scDART_integrated_seurat.rds")
```


